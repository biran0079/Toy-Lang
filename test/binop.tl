print(1+2);
print(3*2);
print(5/2);
print(4-2-3);
print(5+2*4);
print(4*2+3*5);
print(4*(2+3)*5);
print((4)*((2)+(3))*(5));
print(1*2*3*4*5*6*7*8*9*10);
