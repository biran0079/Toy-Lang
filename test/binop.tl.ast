(stmts (call (id print) (exps (+ (int 1) (int 2)))) (call (id print) (exps (* (int 3) (int 2)))) (call (id print) (exps (/ (int 5) (int 2)))) (call (id print) (exps (- (- (int 4) (int 2)) (int 3)))) (call (id print) (exps (+ (int 5) (* (int 2) (int 4))))) (call (id print) (exps (+ (* (int 4) (int 2)) (* (int 3) (int 5))))) (call (id print) (exps (* (* (int 4) (+ (int 2) (int 3))) (int 5)))) (call (id print) (exps (* (* (int 4) (+ (int 2) (int 3))) (int 5)))) (call (id print) (exps (* (* (* (* (* (* (* (* (* (int 1) (int 2)) (int 3)) (int 4)) (int 5)) (int 6)) (int 7)) (int 8)) (int 9)) (int 10)))) (call (id print) (exps (% (int 5) (int 2)))) (call (id print) (exps (% (int 10) (int 5)))) (call (id print) (exps (== (% (int 6) (int 3)) (int 0)))))
