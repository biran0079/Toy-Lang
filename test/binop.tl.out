3
6
2
-1
13
23
100
100
3628800
1
0
1
